src/refdb.cpp

```diff
@@ -80,23 +93,38 @@ namespace referral
         return {};
     }
 
-    MaybeReferral ReferralsViewDB::GetReferral(const std::string& alias) const
+    MaybeReferral ReferralsViewDB::GetReferral(
+            const std::string& alias,
+            int blockheight,
+            const Consensus::Params& params) const
     {
         if (alias.size() == 0 || alias.size() > MAX_ALIAS_LENGTH) {
             return {};
         }
 
         Address address;
-        if (m_db.Read(std::make_pair(DB_ALIAS, alias), address)) {
+
+        auto normalized_alias = alias;
+
+        if (blockheight >= params.safer_alias_blockheight) {
```

Comment:
maybe it's better not to pass blockheight and params here to check if it should be normalized and just pass a `use_normalisez` param? looks like it's not the responsibility of this class to check normalisation conditions